numpy/core/src/multiarray/buffer.c

```diff
@@ -832,11 +832,6 @@ gentype_getbuffer(PyObject *self, Py_buffer *view, int flags)
     descr = PyArray_DescrFromScalar(self);
     view->buf = (void *)scalar_value(self, descr);
     elsize = descr->elsize;
-#ifndef Py_UNICODE_WIDE
-    if (descr->type_num == NPY_UNICODE) {
-        elsize >>= 1;
-    }
-#endif
```

Comment:
This code is unreachable for unicode anyway, it seems - but this change still makes it more correct

Reply:
Update: by adding the buffer interface to Unicode in the second commit, this became reachable.